### DISCUSSION TITLE
Why is this filter not working?

### DISCUSSION THREAD

**Discussion by serkan-bayram** [pocketbase/pocketbase#3684] [Q&A] [answered]
Here is the code: 

```jss
    const records = await pb.collection("ratings").getFullList({
      sort: "-updated",
      filter: `menuId=${menuId}`,
    });

    console.log(records);
```

Here is the error: 
![image](https://github.com/pocketbase/pocketbase/assets/68502470/a75070e7-01e7-4f8d-b182-951769267342)

Menus collection:

![image](https://github.com/pocketbase/pocketbase/assets/68502470/3d9435c6-f0d7-4218-8f59-9c55906ed74d)

Ratings collection:

![image](https://github.com/pocketbase/pocketbase/assets/68502470/42e9c441-f51d-4ede-a046-e235fdc91555)

**Comment by ganigeorgiev** [comment, marked as answer]
The string variable `menuId` must be wrapped in single or double quotes, otherwise it will be treated as an identifier (aka. field name).

```js
const records = await pb.collection("ratings").getFullList({
    // or using the placeholder filter syntax available with v0.19.0+:
    // pb.filter("menuId={:id}", { id: menuId })
    // (see https://github.com/pocketbase/js-sdk#binding-filter-parameters)
    filter: `menuId="${menuId}"`,
});
```

**Reply by serkan-bayram** [reply]
This didn't work. 

```js
filter: `menuId="${menuId}"`,
```
and since I'm not working with v0.19.0+ I can't try to other way, any ideas?

**Reply by ganigeorgiev** [reply]
It should work. Make sure that the request satisfies the List API rule of the collection.

**Reply by serkan-bayram** [reply]
I don't know what I'm thinking, it's right there the column is named "menu" but I'm trying to find it as "menuId", sorry for taking your time. Thanks.

![image](https://github.com/pocketbase/pocketbase/assets/68502470/917e8862-58ac-4699-9bb0-15a76d0d81cd)
